reference-implementation/lib/helpers.js

```diff
@@ -77,7 +77,7 @@ exports.PromiseInvokeOrNoop = (O, P, args) => {
   }
 };
 
-exports.PromiseInvokeOrFallbackOrNoop = (O, P1, args1, P2, args2) => {
+exports.PromiseInvokeOrFallback = (O, P1, args1, F, args2) => {
```

Comment:
This operation needs a different name to reflect its unique semantics.

How about `PromiseInvokeOrCallFallback` ?